core/src/main/java/org/mapfish/print/servlet/job/impl/hibernate/HibernateJobQueue.java

```diff
@@ -209,11 +214,26 @@ public final void shutdown() {
 
   private void cleanup() {
     TransactionTemplate tmpl = new TransactionTemplate(this.txManager);
-    final int nbDeleted =
-        tmpl.execute(
-            status ->
-                HibernateJobQueue.this.dao.deleteOld(
-                    System.currentTimeMillis() - getTimeToKeepAfterAccessInMillis()));
-    metricRegistry.counter(getClass().getName() + ".deleted_old").inc(nbDeleted);
+    String name = MetricRegistry.name(getClass().getSimpleName(), "deletedOldPrintJobs");
+    try (Timer.Context deletion = metricRegistry.timer(name).time()) {
+      final Integer nbDeleted =
+          tmpl.execute(
+              status -> {
+                try {
+                  return HibernateJobQueue.this.dao.deleteOld(
+                      System.currentTimeMillis() - getTimeToKeepAfterAccessInMillis());
+                } catch (Exception e) {
+                  unhealthyCountersHealthCheck.recordUnhealthyProblem(
+                      name, "failedToDeleteOldJobs");
+                  return null;
+                }
+              });
+      if (deletion.stop() > CLEAN_UP_INTERVAL_TIME_UNIT.toNanos(cleanupInterval)) {
+        unhealthyCountersHealthCheck.recordUnhealthyProblem(name, "tooManyJobsToDeleteInInterval");
+      }
+      if (nbDeleted != null && nbDeleted > 0) {
+        metricRegistry.counter(name + ".totalDeleted").inc(nbDeleted);
```

Comment:
```suggestion
        metricRegistry.counter(MetricRegistry.name(name, "deletedJobs")).inc(nbDeleted);
```

Reply:
This suggested change would create the counter:
HibernateJobQueue.deletedOldPrintJobs.deletedJobs
it is not a counter for the number of Jobs deleted each time it is called. It is counter for the total number of jobs deleted. So I suggest
HibernateJobQueue.deletedOldPrintJobs.totalCount ?

Reply:
Why not but what about just HibernateJobQueue.deletedOldPrintJobs.count ?

Reply:
I am trying to highlight this is the total number of deleted jobs. It is cumulative. Maybe cumulativeCount ?
It is to make the difference with what just got deleted the last time the counter was updated and what was deleted by this server instance since it was started.

Reply:
When we see that it's a complete pleonasm!
https://prometheus.io/docs/concepts/metric_types/

Reply:
It is indeed in a pleonasm in the context of Prometheus but not in the English language, where a counter keep counts of anything.
Here are some extracts from the [Metrics Lib](https://metrics.dropwizard.io/4.2.0/getting-started.html):
A gauge is an instantaneous measurement of a value
A counter is just a gauge for an AtomicLong instance. You can increment or decrement its value. For example, we may want a more efficient way of measuring the pending job in a queue. Every time this counter is measured, it will return the number of jobs in the queue.
In our case from the name: HibernateJobQueue.deletedOldPrintJobs.count = 5, we cannot make the difference between "5 print jobs just got deleted" and "the total number of print jobs deleted by this instance = 5" So I am trying to remove this ambiguity. I am not a big fan of cumulative. How about totalCountByInstance ?

Reply:
'can increment or decrement' => in my understanding, it's the same as saying that's cumulatif.
The instant number of deleted job is unusable as a metric (for gauge, possibly a histogram can be use)
=> should be removed, this can be clarified in a documentation that's currently a relay missing point.